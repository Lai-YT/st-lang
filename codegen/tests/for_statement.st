put "increasing, no id:"
for : 0..5
  put "." ..
end for
skip
put "^^^^^^ (6 dots)"

put "decreasing, no id:"
for decreasing : 5..0
  put "." ..
end for
skip
put "^^^^^^ (6 dots)"
