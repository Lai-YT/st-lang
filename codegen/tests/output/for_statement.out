increasing, no id:
......
^^^^^^ (6 dots)
decreasing, no id:
......
^^^^^^ (6 dots)
increasing, with id:
......
^^^^^^ (6 dots)
decreasing, with id:
......
^^^^^^ (6 dots)
nested:
x6
  ......
  ^^^^^^ (6 dots)
  ......
  ^^^^^^ (6 dots)
  ......
  ^^^^^^ (6 dots)
  ......
  ^^^^^^ (6 dots)
  ......
  ^^^^^^ (6 dots)
  ......
  ^^^^^^ (6 dots)
